modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/toolbarbuttons/plugin.js

```diff
@@ -49,6 +59,76 @@
 				title: editor.lang.link.title,
 			});
 
+			editor.ui.addBalloonToolbarButton('Twitter', {
+
+				//
+
+				_MAX_TWEET_LENGTH: 280,
+				_getHref(url, via) {
+					const selectedText = editor
+						.getSelection()
+						.getSelectedText()
+						.substring(0, this._MAX_TWEET_LENGTH);
+
+					// TODO - These parameters need to be revisited to see when to use `url` and `via`.
+					// There is a doc about it in https://developer.twitter.com/en/docs/twitter-for-websites/tweet-button/guides/parameter-reference1
+					// Where:
+					// url -> URL included with the Tweet.
+					// via -> Attribute the source of a Tweet to a Twitter username.(a.k.a the twitter handle)
+
+					// const url = this.props.url;
+					// const via = this.props.via;
```

Comment:
In AlloyEditor's code, there is two props in the component called `via` and `url`. There is a doc about it in https://developer.twitter.com/en/docs/twitter-for-websites/tweet-button/guides/parameter-reference1 and it can be helpful when providing a link to the tweet in case of url and if somebody want to add their/other handle in Twitter. However, when playing around Blog's contentEditor I can't find these values being defined.

Do we have a specific use case for it or it was an unused AlloyEditor's feature?

/cc @boton @adolfopa

Reply:
>Do we have a specific use case for it or it was an unused AlloyEditor's feature?

After a little research, AFAIK these values have never been implemented in Blogs. 

Maybe isn't a bad idea to implement at least the `url` value because I'm not sure how useful is a tweet without context.

Reply:
> Maybe isn't a bad idea to implement at least the url value because I'm not sure how useful is a tweet without context.

Yes, I was trying to imagine how things can be inserted and from an UX POV(I'm not a designer, so, please, consider that I'm just wondering) it requires:

1. We can pass props.url from the published blog URL. However, URLs in DXP are huge and it can affect the maximum number of characters in a tweet.

2. Okay, We decided to implement it. Do We have a plan for `via`? I think we can create a panel or something like to input user's twitter handle.

Just passing an URL can be easy but for second one will require more investigation and a side task to implement it(missing React way to create UIs)

Would you include someone from design in this thread?

Reply:
Hey @boton, while implementing this button I considered the usage a bit weird. I usually select and then share things in readOnly mode, not in editor/builder. What do you think about it?

Reply:
Yep, I'm with you; I think that usage is weird. But on the other hand, I found this [comment in Jira](https://issues.liferay.com/browse/LPS-79209?focusedCommentId=1309490&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1309490) defending it. 

I'm more with your opinion of letting the reader decide what to share and [this feature request](https://issues.liferay.com/browse/LPS-80607) too.

I agree with all of you of drop this button, but we will wait for Pablo opinion 😄 

> We can pass props.url from the published blog URL. However, URLs in DXP are huge and it can affect the maximum number of characters in a tweet.

If we decide to implement it I think that URL will be shortened by Twitter.

> Okay, We decided to implement it. Do We have a plan for via? I think we can create a panel or something like to input user's twitter handle. If we don't want to create another panel we could reuse from user's configuration model(IDK if exists a field for twitter handle but would be a good idea) :P

I honestly have no idea